src/rules/eva-voice-biometric.js

```diff
@@ -0,0 +1,147 @@
+/**
+ *	@title EVA Voice Biometric connector
+ *	@overview EVA Voice Biometric connector rule for Auth0 enables voice enrolment and verification as a second factor
+ *	@gallery true
+ *	@category marketplace
+ *
+ *
+ *	all configuration items are optional:
+ *	AURAYA_URL = optional. EVA endpoint, typically: https://eva-web.mydomain.com/server/oauth
+ *	AURAYA_CLIENT_ID = optional. JWT client id on the EVA server (and this server)
+ *	AURAYA_CLIENT_SECRET = optional. JWT client secret on the EVA server (and this server)
+ *	AURAYA_ISSUER = optional. this app (or "issuer")
+ *
+ *	AURAYA_RANDOM_DIGITS = optional. true|false whether to prompt for random digits
+ *	AURAYA_COMMON_DIGITS = optional. true|false whether to prompt for common digits
+ *	AURAYA_PERSONAL_DIGITS = optional. a user.user_metadata property that contains digits such as phone_number
+ *	AURAYA_COMMON_DIGITS_PROMPT = optional. a digit string to prompt for common digits (e.g '987654321')
+ *	AURAYA_PERSONAL_DIGITS_PROMPT = optional. a string to prompt for personal digits (e.g 'your cell number')
+ *
+ *	AURAYA_DEBUG = optional. if set, controls detailed debug output
+ */
+function evaVoiceBiometric(user, context, callback) {
+
+  const debug = typeof configuration.AURAYA_DEBUG !== 'undefined';
+  if (debug) {
+    console.log(user);
+    console.log(context);
+    console.log(configuration);
+  }
+
+  const eva_url = configuration.AURAYA_URL || 'https://eval-eva-web.aurayasystems.com/server/oauth';
+  const clientSecret = configuration.AURAYA_CLIENT_SECRET || 'o4X0LFKi2caP5ipUwaF4B27cZmfOIh0JXnqmfiC4mHkVskSzbp72Emk3AB6';
+  const clientId = configuration.AURAYA_CLIENT_ID || 'auraya';
+  const issuer = configuration.AURAYA_ISSUER || 'issuer';
+  
+  
+  // Prepare user's enrolment status
+  user.user_metadata = user.user_metadata || {};
+  user.user_metadata.auraya_eva = user.user_metadata.auraya_eva || {};
+
+  // User has initiated a login and is prompted to use voice biometrics
+  // Send user's information and query params in a JWT to avoid tampering
+  function createToken(user) {
+    const options = {
+      expiresInMinutes: 2,
```

Comment:
This is expiration just for the redirect out? In other words, if the voice check takes 3 minutes. this won't fail completely?

Reply:
Correct, it just for the outbound redirect and to account for clock skew. A new JWT is created by EVA for the return callback.

Reply:
However, there is a (configurable) timeout for each audio prompt of about 1 minute within the EVA widget. This is to guard against hacking attempts.